## 2. Contingency tables

We have already seen some contingency tables and its uses in calculating different types of probabilities, lets explore code to produce and analyze contingency tables in R.



First lets define joint, marginal, and conditional probabilities:

### Joint probabilities: Are the chances of two events occurring at the same time p(A$\bigcap$B) or P(A and B), the intersection of two events, these can be mutually exclusive or not mutually exclusive. 

### Marginal probability: Is the probability of occurring one single event, this can be calculated by adding all joint probabilities for that event. 

### Conditional Probabilities: As we have seen, conditional probabilities is the probability of event A occurring given event B.

The equation: $$P(A|B) = \frac{P(A and B)}{P(B)}$$

Lets create a contingency table in R, for this we will use a different way of writing code in R. The package dyplr is a very useful package for data analysis, here is the introductory vignette 

[https://cran.r-project.org/web/packages/dplyr/vignettes/dplyr.html](https://cran.r-project.org/web/packages/dplyr/vignettes/dplyr.html)

This exercise was modified from [http://tinyheero.github.io/2016/03/20/basic-prob.html](http://tinyheero.github.io/2016/03/20/basic-prob.html)

Other examples see
[https://www.youtube.com/watch?time_continue=236&v=pIfpHdGVwLU](https://www.youtube.com/watch?time_continue=236&v=pIfpHdGVwLU)

### Remember, it is extremely important to know the direction of the conditional probability it is not the same to have P(A|B) P(is going to rain given that it is spring) than P(B|A) P(It is spring given that it is going to rain).

Always ask, which variable is hidden? which is out prior information (observed variable), What is the direction of the probability.

### Let's use the attached dataset BCHI2.csv in order to explore contigency tables and multiple probabilities calculations

## The Big cities Health Coalition, maintains a set of databases from multiple issues related to health and safety the country's largest metropolitan area. see [http://www.bigcitieshealth.org/](http://www.bigcitieshealth.org/).

### Today we will using a dataset that explores the major components of opioid addiction and unintentional opioid overdoses. This dataset contains more than 50 health status indicators, death rates, socio-economic and demographic factors. This dataset includes multiple years

In [2]:
install.packages("dplyr")


The downloaded binary packages are in
	/var/folders/79/jxb90vv11gvb4bfw9cs5_kg00000gn/T//RtmpLjU5Pp/downloaded_packages


In [1]:
library("ggplot2")
library("dplyr")
library("reshape2")
library("knitr")


Attaching package: ‘dplyr’

The following objects are masked from ‘package:stats’:

    filter, lag

The following objects are masked from ‘package:base’:

    intersect, setdiff, setequal, union



In [3]:
BCHI = read.csv(file = "BCHI2.csv")

In [7]:
BCHI

Indicator.Category               
1     Behavioral Health/Substance Abuse
2     Behavioral Health/Substance Abuse
3     Behavioral Health/Substance Abuse
4     Behavioral Health/Substance Abuse
5     Behavioral Health/Substance Abuse
6     Behavioral Health/Substance Abuse
7     Behavioral Health/Substance Abuse
8     Behavioral Health/Substance Abuse
9     Behavioral Health/Substance Abuse
10    Behavioral Health/Substance Abuse
11    Behavioral Health/Substance Abuse
12    Behavioral Health/Substance Abuse
13    Behavioral Health/Substance Abuse
14    Behavioral Health/Substance Abuse
15    Behavioral Health/Substance Abuse
16    Behavioral Health/Substance Abuse
17    Behavioral Health/Substance Abuse
18    Behavioral Health/Substance Abuse
19    Behavioral Health/Substance Abuse
20    Cancer                           
21    Chronic Disease                  
22    Chronic Disease                  
23    Chronic Disease                  
24    Chronic Disease                  
25    Chronic Disease                  
26    Chronic Disease                  
27    Chronic Disease                  
28    Chronic Disease                  
29    Chronic Disease                  
30    Chronic Disease                  
⋮     ⋮                                
17464 Maternal and Child Health        
17465 Maternal and Child Health        
17466 Maternal and Child Health        
17467 Maternal and Child Health        
17468 Maternal and Child Health        
17469 Maternal and Child Health        
17470 Maternal and Child Health        
17471 Maternal and Child Health        
17472 Maternal and Child Health        
17473 Maternal and Child Health        
17474 Maternal and Child Health        
17475 Maternal and Child Health        
17476 Maternal and Child Health        
17477 Maternal and Child Health        
17478 Maternal and Child Health        
17479 Maternal and Child Health        
17480 Maternal and Child Health        
17481 Maternal and Child Health        
17482 Maternal and Child Health        
17483 Maternal and Child Health        
17484 Maternal and Child Health        
17485 Maternal and Child Health        
17486 Maternal and Child Health        
17487 Maternal and Child Health        
17488 Maternal and Child Health        
17489 Maternal and Child Health        
17490 Maternal and Child Health        
17491 Maternal and Child Health        
17492 Maternal and Child Health        
17493 Maternal and Child Health        
      Indicator                                                            
1     Percent of Adults Who Binge Drank                                    
2     Percent of Adults Who Binge Drank                                    
3     Percent of Adults Who Binge Drank                                    
4     Percent of Adults Who Binge Drank                                    
5     Percent of Adults Who Binge Drank                                    
6     Percent of Adults Who Binge Drank                                    
7     Percent of Adults Who Binge Drank                                    
8     Percent of Adults Who Binge Drank                                    
9     Percent of Adults Who Binge Drank                                    
10    Percent of Adults Who Binge Drank                                    
11    Percent of Adults Who Binge Drank                                    
12    Percent of Adults Who Binge Drank                                    
13    Percent of Adults Who Binge Drank                                    
14    Percent of Adults Who Binge Drank                                    
15    Percent of Adults Who Binge Drank                                    
16    Percent of High School Students Who Binge Drank                      
17    Percent of High School Students Who Binge Drank                      
18    Percent of High School Students Who Binge Drank                      
19    Percent of High School Students Who Binge Drank                      
20    

In [4]:
BCHI.sex.region.df <-
  BCHI %>%
  group_by(Sex, Region) %>%
  filter(Sex != "Both") %>%
  summarize(n = n())

In [5]:
BCHI.sex.region.df %>%
  dcast(Sex ~ Region, value.nar = "n") %>%
  kable(align = "l", format = "markdown",
        table.attr='class="table table-striped table-hover"')

Using n as value column: use value.var to override.




|Sex    |East |West |
|:------|:----|:----|
|Female |1580 |1471 |
|Male   |1137 |1056 |

|Sex    |East |West |
|:------|:----|:----|
|Female |1580 |1471 |
|Male   |1137 |1056 |

## Lets calculate now that joint probabilities:
which is the probability of two different events occurring at the same time

In [6]:
BCHI.sex.region.prop.df <- 
  BCHI.sex.region.df %>%
  ungroup() %>%
  mutate(prop = n / sum(n))

BCHI.sex.region.prop.df %>%
  dcast(Sex ~ Region, value.var = "prop") %>%
  kable(align = "l", format = "markdown", 
        table.attr = 'class="table table-striped table-hover"')



|Sex    |East      |West      |
|:------|:---------|:---------|
|Female |0.3012967 |0.2805111 |
|Male   |0.2168192 |0.2013730 |

Joint probabilities can be calculated by taking the proportion of times a specific Sex-Region combination occurs divided by total number of all sex-regions combinations (i.e. frequency):

### What can we say about this trend?

## How about the Marginal probabilities:
Marginal probabilities focuses on single events, to calculate marginal probabilities we let constant our random variable we are interested for example: Males, and we sum all of the joint probabilities of the random variable Regions.

In [8]:
Sex.marginal.df <- 
  BCHI.sex.region.prop.df %>%
  group_by(Sex) %>%
  summarize(marginal = sum(prop))

Region.marginal.df <- 
  BCHI.sex.region.prop.df %>%
  group_by(Region) %>%
  summarize(marginal = sum(prop))

In [9]:
BCHI.sex.region.prop.df %>%
  dcast(Sex ~ Region, value.var = "prop") %>%
  left_join(Sex.marginal.df, by = "Sex") %>%
  bind_rows(
    Region.marginal.df %>%
      mutate(Sex = "marginal") %>%
      dcast(Sex ~ Region, value.var = "marginal")
  ) %>%
  kable(align = "l", format = "markdown",
        table.attr = 'class="table table-striped table-hover"')

Warning message in bind_rows_(x, .id):
“binding factor and character vector, coercing into character vector”Warning message in bind_rows_(x, .id):
“binding character and factor vector, coercing into character vector”



|Sex      |East      |West      |marginal  |
|:--------|:---------|:---------|:---------|
|Female   |0.3012967 |0.2805111 |0.5818078 |
|Male     |0.2168192 |0.2013730 |0.4181922 |
|marginal |0.5181159 |0.4818841 |NA        |

|Sex      |East      |West      |marginal  |
|:--------|:---------|:---------|:---------|
|Female   |0.3012967 |0.2805111 |0.5818078 |
|Male     |0.2168192 |0.2013730 |0.4181922 |
|marginal |0.5181159 |0.4818841 |NA        |

### Conditional Probabilities

Lets calculate the conditional probability that Females given that they are from the East coast. lets review the equation:

$$P(Female|East) = \frac{P(Females, East)}{P(East)}$$

P(Females,East) becomes the joint probability and P(East) is our marginal probability of female addiction in the East region.

In [ ]:
joint.prob <- 
  BCHI.sex.region.prop.df %>%
  filter(Sex == "Female", Region == "East") %>%
  .$prop

marg.prob <- 
  Region.marginal.df %>%
  filter(Region == "East") %>%
  .$marginal

cond.prob <- joint.prob / marg.prob
cond.prob